dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/search/index/SolrIndexServiceBean.java

```diff
@@ -22,491 +14,131 @@
-                dvObjectService.updatePermissionIndexTime(dvObject);
-            }
-            return new IndexResponse("indexed all permissions");
+            persistToSolr(definitionPoints);
+            dvObjectService.updatePermissionIndexTime(dvObject.getId());
+            
+            return new IndexResponse("attempted to index permissions for DvObject " + dvObject.getId() + " and update permission index time was sucessfull");
         } catch (SolrServerException | IOException ex) {
             return new IndexResponse("problem indexing");
         }
 
     }
 
-    public IndexResponse indexPermissionsForOneDvObject(DvObject dvObject) {
-        if (dvObject == null) {
-            return new IndexResponse("problem indexing... null DvObject passed in");
-        }
-        long dvObjectId = dvObject.getId();
-        Collection<SolrInputDocument> docs = new ArrayList<>();
+    public IndexResponse indexAllPermissions() {
 
-        List<DvObjectSolrDoc> definitionPoints = determineSolrDocs(dvObject);
+        List<PermissionsSolrDoc> definitionPoints = solrDocFactory.determinePermissionsDocsOnAll();
+        Set<Long> affectedDvObjectIds = collectDvObjectIds(definitionPoints);
 
-        for (DvObjectSolrDoc dvObjectSolrDoc : definitionPoints) {
-            SolrInputDocument solrInputDocument = SearchUtil.createSolrDoc(dvObjectSolrDoc);
-            docs.add(solrInputDocument);
-        }
         try {
-            persistToSolr(docs);
-            boolean updatePermissionTimeSuccessful = false;
-            if (dvObject != null) {
-                DvObject savedDvObject = dvObjectService.updatePermissionIndexTime(dvObject);
-                if (savedDvObject != null) {
-                    updatePermissionTimeSuccessful = true;
-                }
+            persistToSolr(definitionPoints);
+            
+            for (Long dvObjectId : affectedDvObjectIds) {
+                dvObjectService.updatePermissionIndexTime(dvObjectId);
             }
-            return new IndexResponse("attempted to index permissions for DvObject " + dvObjectId + " and updatePermissionTimeSuccessful was " + updatePermissionTimeSuccessful);
+            return new IndexResponse("indexed all permissions");
         } catch (SolrServerException | IOException ex) {
             return new IndexResponse("problem indexing");
         }
 
     }
 
-    private void persistToSolr(Collection<SolrInputDocument> docs) throws SolrServerException, IOException {
+    private void persistToSolr(Collection<PermissionsSolrDoc> docs) throws SolrServerException, IOException {
         if (docs.isEmpty()) {
             // This method is routinely called with an empty list of docs.
             logger.fine("nothing to persist");
             return;
         }
         logger.fine("persisting to Solr...");
+        List<SolrInputDocument> inputDocs = new ArrayList<>();
+        docs.forEach(d -> inputDocs.add(SearchUtil.createSolrDoc(d)));
```

Comment:
Imo lambda variables should be treated like normal variables i.e. they should have meaningful names.